x-pack/plugins/fleet/public/hooks/use_get_agent_incoming_data.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useEffect, useState, useMemo } from 'react';
+
+import type { IncomingDataList } from '../../common/types/rest_spec/agent';
+
+import { sendGetAgentIncomingData } from './index';
+
+export const useGetAgentIncomingData = (agentsIds: string[]) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [incomingData, setIncomingData] = useState<IncomingDataList[]>([]);
+
+  useEffect(() => {
+    const getIncomingData = async () => {
+      const { data } = await sendGetAgentIncomingData({ agentsIds });
+      if (data?.items) {
+        setIncomingData(data?.items);
+        setIsLoading(false);
+      }
+    };
+    if (agentsIds) {
+      getIncomingData();
+    }
+  }, [agentsIds]);
+
+  const enrolledAgents = useMemo(() => incomingData.length, [incomingData.length]);
+  const agentsWithData = useMemo(
```

Comment:
(nit) Could we rename this to indicate that it's a count of agents, and not an array of agent objects? Maybe something like `numAgentsWithData` or `activeAgentCount` or something along those?